### PR TITLE
Update generator to be more robust

### PR DESCRIPTION
**Pull request by MitchellGRead**
This PR is the start of our service refactoring predominently for handling connections and responses.

A client is used to do all handling of the actual get and post commands along with managing the aiohttp client. In the event that something goes wrong the system will retry the request for a default of 5 attempts with each new request done using a new aiohttp client

### CODE REVIEW

File: WorkloadGenerator/docker_context/Client.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        status = self.SERVICE_UNAVAILABLE_STATUS
+        retry_count = 1
+        while retry_count <= max_retry:
+            try:
+                async with self.client.post(url, json=json) as resp:
+                    result, status = await self.getJson(resp)
+
+                if result is None:
+                    logger.error(f'No result received from {url}. Retry {retry_count} of {max_retry}')
+                    retry_count += 1
```
Comment (a-mon) [inline WorkloadGenerator/docker_context/Client.py]: I think this is not needed here. You are already incrementing further down.

File: WorkloadGenerator/docker_context/Client.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                else:
+                    break
```
Comment (a-mon) [inline WorkloadGenerator/docker_context/Client.py]: Why not just return result, status here as well? You seem to be doing this in the last line of the function but doing it here will make it consistent with the other function above.

File: WorkloadGenerator/docker_context/UserCommands.py [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+
+if config.RUN_DEBUG:
+    logging.getLogger().setLevel(logging.DEBUG)
+    logging.basicConfig(
+        format='%(asctime)s - %(levelname)s - %(message)s',
+        datefmt='%d-%b-%y %H:%M:%S'
+    )
+else:
+    logging.disable(logging.DEBUG)
+
```
Comment (a-mon) [inline WorkloadGenerator/docker_context/UserCommands.py]: Can we not use eventLogger.py here as well?

**Review by a-mon — CHANGES_REQUESTED** [review]

File: WorkloadGenerator/docker_context/Client.py (hunk 2 above)
Comment (MitchellGRead) [reply WorkloadGenerator/docker_context/Client.py]: missed changing it over thanks

**Review by a-mon — APPROVED** [review]

**Pull request merged**